cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```
@@ -122,4 +253,372 @@ private String getProgramScheduleId(Id.Program program, SchedulableProgramType p
+    private final Id.Stream streamId;
+    private final AtomicInteger activeTasks;
+
+    private Cancellable notificationSubscription;
+    private ScheduledFuture<?> scheduledPolling;
+    private StreamSizeNotification lastNotification;
+
+    private StreamSubscriber(Id.Stream streamId) {
+      this.streamId = streamId;
+      this.scheduleTasks = Maps.newConcurrentMap();
+      this.lastNotificationLock = new Object();
+      this.activeTasks = new AtomicInteger(0);
+    }
+
+    public void start() throws NotificationFeedException, NotificationFeedNotFoundException {
+      notificationSubscription = notificationService.subscribe(getFeed(), this, notificationExecutor);
+    }
+
+    @Override
+    public void cancel() {
+      if (scheduledPolling != null) {
+        scheduledPolling.cancel(true);
+      }
+      if (notificationSubscription != null) {
+        notificationSubscription.cancel();
+      }
+    }
+
+    /**
+     * Add a new scheduling task based on the data received by the stream referenced by {@code this} object.
+     */
+    public void createScheduleTask(Id.Program programId, SchedulableProgramType programType,
+                                   StreamSizeSchedule streamSizeSchedule, boolean active,
+                                   long baseRunSize, long baseRunTs, boolean persist) {
+      // TODO add a createScheduleTasks, so that if we create multiple schedules for the same stream at the same
+      // time, we don't have to poll the stream many times
+
+      StreamSizeScheduleTask newTask = new StreamSizeScheduleTask(programId, programType, streamSizeSchedule);
+      StreamSizeScheduleTask previous =
+        scheduleTasks.putIfAbsent(getScheduleId(programId, programType, streamSizeSchedule.getName()),
+                                  newTask);
+      if (previous != null) {
+        // We cannot replace an existing schedule - that functionality is not wanted - yet
+        return;
```

so we silently do nothing? Don't you have to return an error? Otherwise the caller thinks it was successful